lib/tools.js

```diff
@@ -91,8 +91,7 @@ function genLog (z) {
     if (p > 1) {
       l = typeof a === 'string' ? countInterp(a, '%j') : 0
       if (l) {
-        n.length = l + countInterp(a, '%d') + countInterp(a, '%s') + 1
```

Comment:
the original reason for truncating arguments to interpolation char count was performance, 
at the time there was a *massive* performance hit from having an argument length greater that interpolation char count in util.format 

this was known behaviour at the time of writing the code, but somewhere along the line the documentation was written without knowledge of this behaviour 

it may not be true any more, we need benchmarks from Node 4, 6, and 8

Reply:
Wouldn't that classify as usage issue? Truncating these arguments is basically just blocking that use case (even inconsistently - depending on placeholder presence, it might cause a silent loss of information).  
I'd argue that regardless of whether it still is costly (which is likely), this would be an explicit usage choice. Just like passing huge JSON objects to be serialized, passing objects without placeholders to be inspected is the user's choice when writing the code. Would you agree?

Reply:
I don't think I agree that it's a usage *choice* - not in the sense of a *conscious choice* where effects are known. That's the difference between this and passing the large objects. With the large objects, you know you're paying that price. With this, things can slow drastically and it's completely unclear why. 

It's been a while, my understanding is that the additional args are mixed into the object (in the case of objects) or concatted to the end at some other point

If that's not the case, I don't think losing info is acceptable either ... but nor is throwing 

But let's see what the benchmarks tell us for 4, 6, 8 – since the simplest solution (yours) may be perfectly fine. If it isn't we need to figure another way around it